custom/enikshay/management/commands/import_voucher_confirmations.py

```diff
@@ -0,0 +1,234 @@
+
+    @property
+    @memoized
+    def all_vouchers_in_domain(self):
+        voucher_ids = self.accessor.get_case_ids_in_domain(CASE_TYPE_VOUCHER)
+        return {
+            voucher.get_case_property(VOUCHER_ID): voucher
+            for voucher in self.accessor.iter_cases(voucher_ids)
+            if voucher.get_case_property('state') in (
+                'approved', 'paid', 'rejected', 'expired', 'cancelled', 'canceled')
+            or voucher.get_case_property('voucher_approval_status') in ('approved', 'partially_approved')
+        }
+
+    def write_csv(self, filename, headers, rows):
+        filename = "voucher_confirmations-{}.csv".format(filename)
+        print "writing {}".format(filename)
+        with open(filename, 'w') as f:
+            writer = csv.writer(f)
+            writer.writerow(headers)
+            writer.writerows(rows)
+
+    def log_voucher_updates(self, voucher_updates):
+        headers = ['ReadableID'] + self.voucher_api_properties + self.voucher_update_properties
+
+        def make_row(voucher_update):
+            api_payload = VoucherPayload.create_voucher_payload(voucher_update.voucher)
+            return [voucher_update.voucher.get_case_property(VOUCHER_ID)] + [
+                api_payload[prop] for prop in self.voucher_api_properties
+            ] + [
+                voucher_update[prop] for prop in self.voucher_update_properties
+            ]
+
+        rows = map(make_row, voucher_updates)
+        self.write_csv('updates', headers, rows)
+
+    def log_unrecognized_vouchers(self, headers, unrecognized_vouchers):
+        self.write_csv('unrecognized', headers, unrecognized_vouchers)
+
+    def log_unmodified_vouchers(self, voucher_ids_to_update):
+        unmodified_vouchers = [
+            voucher for voucher_id, voucher in self.all_vouchers_in_domain.items()
+            if voucher_id not in voucher_ids_to_update
+        ]
+        headers = ['ReadableID', 'URL', 'state', 'voucher_approval_status'] + self.voucher_api_properties
+
+        def make_row(voucher):
+            api_payload = VoucherPayload.create_voucher_payload(voucher)
+            return [
+                voucher.get_case_property(VOUCHER_ID),
+                'https://enikshay.in/a/enikshay/reports/case_data/{}'.format(voucher.case_id),
+                voucher.get_case_property('state'),
+                voucher.get_case_property('voucher_approval_status'),
+            ] + [
+                api_payload[prop] for prop in self.voucher_api_properties
+            ]
+
+        rows = map(make_row, unmodified_vouchers)
+        self.write_csv('updates', headers, rows)
+
+    def update_vouchers(self, voucher_updates):
+        print "updating voucher cases"
+        for chunk in chunked(with_progress_bar(voucher_updates), 100):
+            updates = [
+                (update.case_id, update.properties, False)
+                for update in chunk
+            ]
+            if self.commit:
```

Comment:
It was actually originally that, but I wanted the non-commit version to run as much as possible (for instance, in case something crazy happened and `chunk` contained things that weren't these update objects).